R/assertions.R

```diff
@@ -1424,3 +1424,51 @@ assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL)
     ))
   }
 }
+
+#' Is a Variable is a Date or Datetime Variable?
+#'
+#' Checks if a variable vector is a date or datetime variable
+#'
+#' @param var The variable to check
+#'
+#' @param var_name The name of the variable. If the argument is specified, the
+#'   specified name is displayed in the error message.
+#'
+#' @return
+#' The function throws an error if `var` is not a date or datetime variable
+#' and returns the input invisibly otherwise.
+#'
+#' @export
+#'
+#' @author Sadchla Mascary
+#'
+#' @keywords assertion
+#'
+#' @examples
+#' library(tibble)
+#' library(rlang)
+#'
+#' example_fun <- function(var) {
+#'   assert_date_vector(var)
+#' }
+#'
+#' my_dates_chr <- c("2018-08-23", "2022-01-30", "1993-07-14")
+#' my_dates_fmt <- as.Date(my_dates_chr, tz = "UTC")
+#'
+#' example_fun(
+#'   var = my_dates_fmt
+#' )
```

Comment:
Would be good to have an example that shows that is fails when you input `my_dates_chr`. See other assertions for examples of how to use `try()`.

Reply:
Also we **never** explicitly name the first argument (see programming strategy).